### ISSUE TITLE
[Question] Click event listener not triggered on push notification when app is force closed

### ISSUE THREAD

**Issue by dipendra-sharma** [OneSignal/react-native-onesignal#1549]
### How can we help?

I am currently using `"react-native-onesignal": "^5.0.0"` for handling push notifications in my app. Everything works as expected when the app is in the background or foreground: the notifications are received and when clicked, the event listener triggers and navigates the user to the appropriate screen.

However, an issue arises when the app is in a force-closed or killed state. Even though the device receives the push notification and the app is launched upon clicking the notification, the click event listener does not get triggered. This results in the user not being navigated to the intended screen.

**Platform:**
- Android
- iOS

**Steps to reproduce:**
1. Force close the app.
2. Send a push notification to the device.
3. Click on the received notification.

**Expected behavior:**
Upon clicking the notification, the app should launch and navigate the user to the appropriate screen as per the event listener.

**Actual behavior:**
The app launches, but the user is not navigated to the intended screen because the click event listener isn't triggered.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jennantilla** [comment]
@dipendra-sharma thanks for reporting! We'll investigate this and get back to you as soon as possible.

**Comment by Bjarkinn** [comment]
This is the same issue as [this one](https://github.com/OneSignal/react-native-onesignal/issues/1547)

We are experiencing the same problem (only iOS) on version 5.0.0 and this is kind of a dealbreaker for us as navigating to a specific screen upon clicking a notification is a crucial UX in our product.

Has this been investigated and can we expect this to be fixed soon?

We previously used `@react-native-firebase/messaging@18.0.0` to handle push notifications and they had the following method that would also be sufficient @jennantilla

```js
    /**
     * When a notification from FCM has triggered the application to open from a quit state,
     * this method will return a `RemoteMessage` containing the notification data, or `null` if
     * the app was opened via another method.
     *
     * See `onNotificationOpenedApp` to subscribe to when the notification is opened when the app
     * is in a background state.
     */
    getInitialNotification(): Promise<RemoteMessage | null>;
```

**Comment by jennantilla** [comment]
@dipendra-sharma @Bjarkinn thank you for your patience!

We have a fix for this in our latest release, [5.0.1](https://github.com/OneSignal/react-native-onesignal/releases/tag/5.0.1). Please let us know if you have any additional questions or concerns!

**Comment by abdulsathar24i** [comment]
Hi @jennantilla 
Could you provide any code snippets or references on how to enable deep linking with navigation when the app is in a closed state for React Native?
Thanks in advance
